src/main/java/com/vmoscalciuc/country/controller/CityController.java

```diff
@@ -3,21 +3,51 @@
 import com.vmoscalciuc.country.dto.CityDtoRequest;
 import com.vmoscalciuc.country.dto.CityDtoResponse;
 import com.vmoscalciuc.country.model.City;
-import com.vmoscalciuc.country.service.GenericService;
+import com.vmoscalciuc.country.service.CityService;
+import com.vmoscalciuc.country.facade.CityFacade;
 import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.io.IOException;
+import java.util.List;
+
 
 @RestController
 @RequestMapping("/api/city")
 public class CityController extends BaseController<City, Long, CityDtoRequest, CityDtoResponse> {
+    private final CityService cityService;
+    private final CityFacade cityFacade;
 
     @Autowired
-    public CityController(GenericService<City, Long, CityDtoRequest, CityDtoResponse> service, ModelMapper modelMapper) {
-        super(service, modelMapper);
+    public CityController(CityService cityService, ModelMapper modelMapper,CityFacade cityFacade) {
+        super(cityService, modelMapper);
+        this.cityService = cityService;
+        this.cityFacade = cityFacade;
+    }
+
+    @GetMapping("/readExcel")
+    @ResponseStatus(code = HttpStatus.OK)
+    public List<CityDtoRequest> getAllEntitiesFromExcel() throws IOException {
+        return cityFacade.readDataFromExcel("C:/Users/VitalicPC/Desktop/cities.xlsx");
```

Comment:
C:/Users/VitalicPC/Desktop/cities.xlsx expose this as a path and inject it , so we will be able to configure it via .properties file.